src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -171,6 +171,10 @@ HeapWord* ShenandoahFreeSet::try_allocate_in(ShenandoahHeapRegion* r, Shenandoah
 
     r->set_affiliation(req.affiliation());
     r->set_update_watermark(r->bottom());
+
+    // Any OLD region allocated during concurrent coalesce-and-fill does not need to be coalesced and filled.
+    // This code is only necessary if req.affiliation() is old, but harmless if not.
+    r->finish_coalesce_and_fill(); 
```

Comment:
Not sure I understand why that's the case. This region could still contain garbage objects that could break the remembered set scan. How does allocating out of the region make these objects parseable?